plasmapy/particles/particle_class.py

```diff
@@ -2023,7 +2022,88 @@ def mass(self, m: u.kg):
+particle_like = Union[str, Integral, Particle, CustomParticle]
+
+particle_like.__doc__ = """
+An `object` is particle-like if it can be identified as an instance of
+`~plasmapy.particles.particle_class.Particle` or
+`~plasmapy.particles.particle_class.CustomParticle`, or cast into one.
+
+Notes
+-----
+Real world particles are typically represented as instances of the
+`~plasmapy.particles.particle_class.Particle` class in PlasmaPy.
+
+>>> from plasmapy.particles import Particle
+
+>>> Particle("proton")
+Particle("p+")
+>>> Particle("neutron")
+Particle("n")
+
+All `~plasmapy.particles.particle_class.Particle` instances and variables
+that can be cast into a `~plasmapy.particles.particle_class.Particle`
+instances are particle-like.
+
+An **element** may also be represented by a string that contains the
+atomic symbol (case-sensitive) or the name of the element, or an integer
+representing the atomic number.  For example, iron can be represented as
+``"Fe"``, ``"iron"``, ``"Iron"``, or ``26``.  If any of these objects is
+passed to `~plasmapy.particles.particle_class.Particle`, it will create
+a `~plasmapy.particles.particle_class.Particle` instance for iron.
+
+An **isotope** may also be represented by a string that contains an
+atomic symbol or element name followed by a hyphen and the mass number
+(with no spaces in between).   For example, :sup:`56`\ Fe can be
+represented as ``"Fe-56"``, ``"iron-56"``, or ``Particle("Fe-56")``.
+A `~plasmapy.particles.particle_class.Particle` instance for an isotope
+can also be created by passing a representation of iron as the first
+argument and specifying the mass number with the argument ``mass_numb``.
+
+>>> Particle(26, mass_numb=56)
+Particle("Fe-56")
+
+An **ion** or **ionic level** is represented as a string that contains a
+representation of an element or isotope, followed by a space, and then
+the charge information (e.g., ``"2+"``).  The charge information in the
+string includes an integer either followed or preceded by a plus or
+minus sign.  For example, Fe:sup:`2+` can be represented as
+``"Fe 2+"``, while :sup:`56`\ Fe:sup:`0+` can be represented as
+``"iron-56 +0"``.  Ions can also be represented using the Roman numeral
+notation that is common in astrophysical spectroscopy.  Here, the Roman
+numeral represents the integer charge plus one (e.g., ``"H I"``
+represents H:sup:`0+` and ``"He-4 II"`` represents :sup:`4`\ He:sup:`1+`).
+The integer charge can alternatively be specified using the ``Z``
+keyword to `~plasmapy.particles.particle_class.Particle`.
+
+>>> Particle("He-4", Z=2)
+Particle("He-4 2+")
+
+A **special particle** may be represented by a string that contains
+the name of the particle (case-insensitive) or a standard symbol for it
+(case-sensitive).  A neutron can be represented as ``"n"`` or
+``"neutron"``; a proton can be represented as ``"p+"``, ``"p"``, or
+``"Proton"``; and an electron can be represented by ``"e-"``, ``"e"``,
+or ``"ELECTRON"``.
```

Comment:
[For future refactoring]  I would say the docstring of `particle_like` shouldn't duplicate, but reference, documentation is other areas.   This  block seems like it should be in the main `Particle` documentation and not here.  However, after looking at the main particle documentation, the documentation does not seem to cleanly cover what is said here.  At some point what is said here should be migrated to the main documentation.